api/rest/client/methods_test.go

```diff
@@ -169,6 +174,105 @@ func TestUnpin(t *testing.T) {
 	testClients(t, api, testF)
 }
 
+type pathCase struct {
+	path    string
+	wantErr bool
+}
+
+var pathTestCases = []pathCase{
+	{
+		test.TestCidResolved,
+		false,
+	},
+	{
+		test.TestPathIPFS1,
+		false,
+	},
+	{
+		test.TestPathIPFS2,
+		false,
+	},
+	{
+		test.TestPathIPNS1,
+		false,
+	},
+	{
+		test.TestPathIPLD1,
+		false,
+	},
+	{
+		test.TestInvalidPath1,
+		true,
+	},
+}
+
+func TestPinPath(t *testing.T) {
+	ctx := context.Background()
+	api := testAPI(t)
+	defer shutdown(api)
+
+	opts := types.PinOptions{
+		ReplicationFactorMin: 6,
+		ReplicationFactorMax: 7,
+		Name:                 "hello there",
+	}
+
+	resultantPin := types.Pin{
```

Comment:
just realized it's safer to do `types.PinWithOpts(cid, opts)` to get a well initialized pin object. It helps things not breaking when we change fields or defaults in the Pin type.

I will mark other places where I see this.